From eeb0c97a6c3a10b6552549b5641f1e60270aa43a Mon Sep 17 00:00:00 2001
From: P Praneesh <quic_ppranees@quicinc.com>
Date: Sat, 10 Aug 2024 18:22:39 +0530
Subject: [PATCH 1/6] QSDK: ath12k: move wmm stats under stats disable

Even though statistics are turned off via debugfs, certain metrics
such as wmm_stats continue to be computed unnecessarily. Move these stats
into stats_disable check.

Patch-Dependency: 653-003-Add-provision-to-configure-rx-hashmap.patch

Signed-off-by: P Praneesh <quic_ppranees@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/dp_rx.c | 33 +++++++++++++++----------
 1 file changed, 20 insertions(+), 13 deletions(-)

diff --git a/drivers/net/wireless/ath/ath12k/dp_rx.c b/drivers/net/wireless/ath/ath12k/dp_rx.c
index 4ec2092..29ae567 100644
--- a/drivers/net/wireless/ath/ath12k/dp_rx.c
+++ b/drivers/net/wireless/ath/ath12k/dp_rx.c
@@ -3655,19 +3655,26 @@ try_again:
 		rxcb->hw_link_id = hw_link_id;
 		rxcb->peer_id = le32_get_bits(desc.rx_mpdu_info.peer_meta_data,
 					      RX_MPDU_DESC_META_DATA_PEER_ID);
-		rxcb->tid = le32_get_bits(desc.rx_mpdu_info.info0,
-					  RX_MPDU_DESC_INFO0_TID);
-		
-		ar->wmm_stats.rx_type = ath12k_tid_to_ac(rxcb->tid > ATH12K_DSCP_PRIORITY ? 0: rxcb->tid);
-
-		if (ath12k_debugfs_is_extd_rx_stats_enabled(ar) && rxcb->peer_id) {
-			rcu_read_lock();
-			peer = ath12k_peer_find_by_peer_list_index(ar, rxcb->peer_id);
-			if (peer && peer->sta)
-				arsta = ath12k_peer_get_link_sta(src_ab, peer);
-			if (arsta)
-				atomic_inc(&arsta->drv_rx_pkts.pkts_frm_hw);
-			rcu_read_unlock();
+
+		if (unlikely(!ab->stats_disable)) {
+			rxcb->tid = le32_get_bits(desc.rx_mpdu_info.info0,
+						  RX_MPDU_DESC_INFO0_TID);
+			ar->wmm_stats.rx_type =
+				ath12k_tid_to_ac(rxcb->tid > ATH12K_DSCP_PRIORITY ? 0 :
+						 rxcb->tid);
+
+			if (ath12k_debugfs_is_extd_rx_stats_enabled(ar) &&
+			    rxcb->peer_id) {
+				rcu_read_lock();
+				peer =
+				ath12k_peer_find_by_peer_list_index(ar,
+								    rxcb->peer_id);
+				if (peer && peer->sta)
+					arsta = ath12k_peer_get_link_sta(src_ab, peer);
+				if (arsta)
+					atomic_inc(&arsta->drv_rx_pkts.pkts_frm_hw);
+				rcu_read_unlock();
+			}
 		}
 
 		__skb_queue_tail(&msdu_list, msdu);
-- 
2.34.1

